redis/src/main/scala/zio/redis/api/Keys.scala

```diff
@@ -13,72 +13,293 @@ import zio.{ Chunk, ZIO }
+   */
+  final def expire(key: String, timeout: Duration): ZIO[RedisExecutor, RedisError, Boolean] = Expire.run((key, timeout))
+
+  /**
+   * Deletes the key at the specific timestamp. A timestamp in the past will delete the key immediately.
+   *
+   * @param key key
+   * @param timestamp an absolute Unix timestamp (seconds since January 1, 1970)
+   * @return true, if the timeout was set, false if the key didn't exist
+   *
+   * @see [[expire]]
+   * @see <a href="https://redis.io/commands/expireat">https://redis.io/commands/expireat</a>
+   */
+  final def expireAt(key: String, timestamp: Instant): ZIO[RedisExecutor, RedisError, Boolean] =
+    ExpireAt.run((key, timestamp))
+
+  /**
+   * Returns all keys matching pattern.
+   *
+   * @param pattern string pattern
+   * @return keys matching pattern
+   *
+   * @see <a href="https://redis.io/commands/keys">https://redis.io/commands/keys</a>
+   */
+  final def keys(pattern: String): ZIO[RedisExecutor, RedisError, Chunk[String]] = Keys.Keys.run(pattern)
+
+  /**
+   * Atomically transfer a key from a source Redis instance to a destination Redis instance. On success the key is deleted
+   * from the original instance and is guaranteed to exist in the target instance.
+   *
+   * @param host remote redis host
+   * @param port remote redis instance port
+   * @param key key to be transferred or empty string if using the keys option
+   * @param destinationDb remote database id
+   * @param timeout timeout in milliseconds
+   * @param auth optionally provide password for the remote instance
+   * @param copy copy option, to not remove the key from the local instance
+   * @param replace replace option, to replace existing key on the remote instance
+   * @param keys keys option, to migrate multiple keys, non empty list of keys
+   * @return string OK on success, or NOKEY if no keys were found in the source instance
+   *
+   * @see <a href="https://redis.io/commands/migrate">https://redis.io/commands/migrate</a>
+   */
   final def migrate(
-    a: String,
-    b: Long,
-    c: String,
-    d: Long,
-    e: Long,
-    f: Option[Copy] = None,
-    g: Option[Replace] = None,
-    h: Option[Auth] = None,
-    i: Option[(String, List[String])]
-  ): ZIO[RedisExecutor, RedisError, String] = Migrate.run((a, b, c, d, e, f, g, h, i))
-
-  final def move(a: String, b: Long): ZIO[RedisExecutor, RedisError, Boolean] = Move.run((a, b))
-
-  final def persist(a: String): ZIO[RedisExecutor, RedisError, Boolean] = Persist.run(a)
-
-  final def pExpire(a: String, b: Duration): ZIO[RedisExecutor, RedisError, Boolean] = PExpire.run((a, b))
-
-  final def pExpireAt(a: String, b: Instant): ZIO[RedisExecutor, RedisError, Boolean] = PExpireAt.run((a, b))
-
-  final def pTtl(a: String): ZIO[RedisExecutor, RedisError, Duration] = PTtl.run(a)
-
+    host: String,
+    port: Long,
+    key: String,
+    destinationDb: Long,
+    timeout: Long,
```

Comment:
possibly change to duration

Reply:
This might be a good idea. I'll double check and we can have it as a separate issue. Wdyt?